yew/src/services/websocket.rs

```diff
@@ -172,13 +180,30 @@ impl WebSocketService {
         &mut self,
         url: &str,
         notification: &Callback<WebSocketStatus>,
-    ) -> Result<ConnectCommon, &str> {
+    ) -> Result<ConnectCommon, WebSocketError> {
         let ws = WebSocket::new(url);
-        if ws.is_err() {
-            return Err("Failed to created websocket with given URL");
-        }
 
-        let ws = ws.map_err(|_| "failed to build websocket")?;
+        let ws = ws.map_err(
+            |#[cfg(feature = "std_web")] _, #[cfg(feature = "web_sys")] ws_error| {
```

Comment:
I think it might be a bit cleaner to put the `cfg` attribute on the closure itself rather than in the params and also in the body, thoughts?

Reply:
I was thinking about that – I think this way reduces the amount of code but I agree that it would be more readable doing it that way.